kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/KafkaRequestHandler.java

```diff
@@ -349,46 +370,17 @@ protected void handleTopicMetadataRequest(KafkaHeaderAndRequest metadataHar,
-                                if (partitionedTopicMetadata.partitions > 0) {
-                                    if (log.isDebugEnabled()) {
-                                        log.debug("Topic {} has {} partitions",
-                                            topic, partitionedTopicMetadata.partitions);
-                                    }
-                                    pulsarTopicNames = IntStream
-                                        .range(0, partitionedTopicMetadata.partitions)
-                                        .mapToObj(i ->
-                                            TopicName.get(pulsarTopicName.toString() + PARTITIONED_TOPIC_SUFFIX + i))
-                                        .collect(Collectors.toList());
-                                    pulsarTopics.put(topic, pulsarTopicNames);
-                                } else {
-                                    if (kafkaConfig.isAllowAutoTopicCreation()) {
-                                        if (log.isDebugEnabled()) {
-                                            log.debug("[{}] Request {}: Topic {} has single partition, "
-                                                    + "auto create partitioned topic",
-                                                ctx.channel(), metadataHar.getHeader(), topic);
-                                        }
-                                        admin.topics().createPartitionedTopicAsync(pulsarTopicName.toString(),
-                                                kafkaConfig.getDefaultNumPartitions());
-                                        final TopicName newTopic = TopicName
-                                            .get(pulsarTopicName.toString() + PARTITIONED_TOPIC_SUFFIX + 0);
-                                        pulsarTopics.put(topic, Lists.newArrayList(newTopic));
-
-                                    } else {
-                                        log.error("[{}] Request {}: Topic {} has single partition, "
-                                                        + "Not allow to auto create partitioned topic",
-                                                ctx.channel(), metadataHar.getHeader(), topic);
-                                        // not allow to auto create topic, return unknown topic
-                                        allTopicMetadata.add(
-                                            new TopicMetadata(
-                                                Errors.UNKNOWN_TOPIC_OR_PARTITION,
-                                                topic,
-                                                false,
-                                                Collections.emptyList()));
-                                    }
+                                int numPartitions = partitionedTopicMetadata.partitions;
+                                if (numPartitions <= 0) {
+                                    throw new RuntimeException("Unexpected numPartitions: " + numPartitions);
```

Comment:
we should not throw exception here. we need to pass the error back to client through response.
as above 
```
  allTopicMetadata.add(
                                            new TopicMetadata(
                                                Errors.UNKNOWN_TOPIC_OR_PARTITION,
                                                topic,
                                                false,
                                                Collections.emptyList()));
```

Reply:
If no test added, we may consider add a test for this error case.

Reply:
Here I'm not sure whether the case should be handled, because the result of `getPartitionedTopicMetadataAsync` couldn't be non-positive. I think it's better to delete these lines here and comment that `numPartitions` must be positive.

Reply:
Make sense. If that is the case, how about adding a verification of the setting "allowAutoTopicCreationType" == partitioned, at the time of kop init?